src/safeposix/syscalls/fs_calls.rs

```diff
@@ -785,43 +846,63 @@ impl Cage {
                         socket_inode_obj.linkcount += 1; //add link to inode
```

Comment:
i am kinda curious if link syscall works on sockets?

Reply:
Updated the comment.